src/posit/connect/resources.py

```diff
@@ -3,46 +3,24 @@
 import posixpath
 import warnings
 from abc import ABC
-from dataclasses import dataclass
 from typing import (
     TYPE_CHECKING,
     Any,
     Iterable,
     Sequence,
 )
 
-from posit.connect.paginator import Paginator
-
 from .context import Context
+from .paginator import Paginator
 
 if TYPE_CHECKING:
-    import requests
-
     from .context import Context
-    from .urls import Url
-
-
-@dataclass(frozen=True)
-class ResourceParameters:
-    """Shared parameter object for resources.
-
-    Attributes
-    ----------
-    session: requests.Session
-        A `requests.Session` object. Provides cookie persistence, connection-pooling, and
-        configuration.
-    url: str
-        The Connect API base URL (e.g., https://connect.example.com/__api__)
-    """
-
-    session: requests.Session
-    url: Url
 
 
 class Resource(dict):
-    def __init__(self, /, params: ResourceParameters, **kwargs):
-        self.params = params
+    def __init__(self, /, ctx: Context, **kwargs):
```

Comment:
This would allow for ctx to never conflict with `kwargs`.
```suggestion
    def __init__(self, ctx: Context, /, **kwargs):
```

Reply:
💯